lms/static/js/learner_dashboard/program_list_factory.js

```diff
@@ -5,16 +5,24 @@
         'js/learner_dashboard/views/collection_list_view',
         'js/learner_dashboard/views/sidebar_view',
         'js/learner_dashboard/views/program_card_view',
+        'js/learner_dashboard/views/empty_programs_view',
         'js/learner_dashboard/collections/program_collection'
     ],
-    function (CollectionListView, SidebarView, ProgramCardView, ProgramCollection) {
+    function (CollectionListView, SidebarView, ProgramCardView, EmptyProgramsView, ProgramCollection) {
         return function (options) {
-            new CollectionListView({
-                el: '.program-cards-container',
-                childView: ProgramCardView,
-                collection: new ProgramCollection(options.programsData)
-            }).render();
 
+            if (options && options.length > 0 && options.programsData) {
+                new CollectionListView({
+                    el: '.program-cards-container',
+                    childView: ProgramCardView,
+                    collection: new ProgramCollection(options.programsData)
+                }).render();
+            } else {
+                new EmptyProgramsView({
+                    el: '.program-list-wrapper',
+                    context: options
+                }).render();
+            }
```

Comment:
I think the approach of adding a View just for when their is no collection to render is not ideal. I think it makes sense for the CollectionListView to be updated to handle the case where the collection is empty, and in that instance render the empty_programs_list template. It seems a common use case for a collection view to need a default state.